### DISCUSSION TITLE
RISC-V support

### DISCUSSION THREAD

**Discussion by andreamancuso** [microsoft/vcpkg#42548] [General]
Hello folks,
I finally got my hands on my Milk-V Mars (RISC-V architecture, 8GB RAM) and have tried the following

> user@milkv:~/dev/xframes/packages/dear-imgui/cpp/tests$ ../deps/vcpkg/bootstrap-vcpkg.sh 
> [...]
> checks.cpp:(.text+0x104): undefined reference to `__atomic_exchange_1'
> collect2: error: ld returned 1 exit status
> [215/216] Building CXX object CMakeFiles/vcpkg-test.dir/src/vcpkg-test/util.cpp.o
> ninja: build stopped: subcommand failed.

Anyone got ideas? I'm happy to donate my board, or help you buy one from arace.tech so you can reproduce the issue

**Comment by andreamancuso** [comment]
Looks like it works out of the box on bianbi on Banana Pi 

https://www.reddit.com/r/RISCV/comments/1h7hmsw/does_vcpkg_compile_for_you/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by ras0219-msft** [comment]
From the original reddit thread, it looks like this is a gcc-13 vs -14 issue; RISCV doesn't have subword atomics, so they're implemented via a runtime library in gcc-13 (`-latomic`). In gcc-14+ it seems the compiler has learned to emit code for it instead of relying a runtime library, so things appear to work.

**Reply by andreamancuso** [reply]
Excellent news!
